merino-cache/src/memory.rs

```diff
@@ -16,13 +16,59 @@ use merino_suggest::{
 };
 use std::{
     borrow::Cow,
+    collections::HashMap,
     fmt::Debug,
     hash::Hash,
     sync::Arc,
     time::{Duration, Instant},
 };
 use tracing::Instrument;
 
+use arc_swap::ArcSwap;
+
+lazy_static! {
+    static ref LOCK_TABLE: ArcSwap<HashMap<String, Instant>> =
+        ArcSwap::from_pointee(HashMap::new());
+}
+
+impl LOCK_TABLE {
```

Comment:
Oh, that's really interesting. I didn't realize you could do that with `lazy_static`.

Reply:
It is a bit of a cheat, but rust allows you to `impl` a surprising number of things.